pkg/account/component_test.go

```diff
@@ -1219,3 +1223,90 @@ func TestRemoveAttributeFromUser(t *testing.T) {
 		assert.Nil(t, err)
 	})
 }
+
+func TestGetLinkedAccounts(t *testing.T) {
+	var mockCtrl = gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	var mocks = createComponentMocks(mockCtrl)
+	var component = mocks.createComponent()
+
+	var accessToken = "TOKEN=="
+	var currentRealm = "test-community"
+	var anyError = errors.New("any error")
+
+	t.Run("Get linked accounts with succces", func(t *testing.T) {
+		var providerAlias = "idp"
+
+		var kcAccountRep = kc.LinkedAccountRepresentation{
+			ProviderAlias: &providerAlias,
+		}
+
+		var kcAccountsRep []kc.LinkedAccountRepresentation
+		kcAccountsRep = append(kcAccountsRep, kcAccountRep)
+
+		mocks.keycloakAccountClient.EXPECT().GetLinkedAccounts(accessToken, currentRealm).Return(kcAccountsRep, nil)
+
+		var ctx = context.WithValue(context.Background(), cs.CtContextAccessToken, accessToken)
+		ctx = context.WithValue(ctx, cs.CtContextRealm, currentRealm)
+
+		apiAccountsRep, err := component.GetLinkedAccounts(ctx)
+
+		var expectedAPIAccountRep = api.LinkedAccountRepresentation{
+			ProviderAlias: &providerAlias,
+		}
+
+		var expectedAPIAccountsRep []api.LinkedAccountRepresentation
+		expectedAPIAccountsRep = append(expectedAPIAccountsRep, expectedAPIAccountRep)
+
+		assert.Nil(t, err)
+		assert.Equal(t, expectedAPIAccountsRep, apiAccountsRep)
+	})
+
+	t.Run("Error", func(t *testing.T) {
+		mocks.keycloakAccountClient.EXPECT().GetLinkedAccounts(accessToken, currentRealm).Return([]kc.LinkedAccountRepresentation{}, anyError)
+
+		var ctx = context.WithValue(context.Background(), cs.CtContextAccessToken, accessToken)
+		ctx = context.WithValue(ctx, cs.CtContextRealm, currentRealm)
+
+		_, err := component.GetLinkedAccounts(ctx)
+
+		assert.NotNil(t, err)
+	})
+}
+
+func TestDeleteLinkedAccount(t *testing.T) {
+	var mockCtrl = gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	var mocks = createComponentMocks(mockCtrl)
+	var accountComponent = mocks.createComponent()
+
+	var accessToken = "TOKEN=="
+	var realmName = "test-community"
+	var username = "username"
+	var userID = "userID"
+	var providerAlias = "idp"
+	var anyError = errors.New("any error")
+
+	var ctx = context.WithValue(context.Background(), cs.CtContextAccessToken, accessToken)
+	ctx = context.WithValue(ctx, cs.CtContextRealm, realmName)
+	ctx = context.WithValue(ctx, cs.CtContextUserID, userID)
+	ctx = context.WithValue(ctx, cs.CtContextUsername, username)
+
+	t.Run("Delete linked account with succces", func(t *testing.T) {
+		mocks.keycloakAccountClient.EXPECT().DeleteLinkedAccount(accessToken, realmName, providerAlias).Return(nil)
```

Comment:
Here, it's easier to understand... Context has been initialized earlier, we quickly see that the difference is the return anyError in the fail case when success case return nil followed by another call to a mock